api/endpoints/skill.py

```diff
@@ -215,6 +216,39 @@ async def list_sub_skills(*, root_skill_id: str, user: User | None = public_auth
     )
 
 
+@router.get("/skilltree/subskills/search", responses=responses(SearchResults))
+@redis_cached("skills", "search_term", "user")
+async def search_sub_skill(*, search_term: str, user: User | None = public_auth) -> Any:
+    """Search for sub skills, root skills, and courses."""
+
+    sub_skills = [
+        sub_skill.serialize
+        async for sub_skill in await db.stream(
+            select(models.SubSkill).where(
+                (models.SubSkill.id.ilike(f"%{search_term}%")) | (models.SubSkill.name.ilike(f"%{search_term}%"))
+            )
+        )
+    ]
+
+    root_skills = [
+        root_skill.serialize
+        async for root_skill in await db.stream(
+            select(models.RootSkill).where(
+                (models.RootSkill.id.ilike(f"%{search_term}%")) | (models.RootSkill.name.ilike(f"%{search_term}%"))
+            )
+        )
+    ]
+
+    courses = [
+        course.serialize
+        async for course in await db.stream(
+            select(models.SkillCourse).where(models.SkillCourse.course_id.ilike(f"%{search_term}%"))
+        )
+    ]
```

Comment:
We should only return the course summary (see the `list_courses` endpoint) since otherwise the result could include the video ids of courses the user has not paid for. Also, do we really have to return the full course (summary) including all sections and lectures? In any case, you probably want to use the `COURSES` constant (again, see `list_courses`) because courses are currently not stored in the database.